### PR TITLE
libcircle: add v0.3

### PR DESCRIPTION
**Pull request by adammoody**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libcircle/package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Libcircle(AutotoolsPackage):
     url      = "https://github.com/hpc/libcircle/releases/download/0.2.1-rc.1/libcircle-0.2.1-rc.1.tar.gz"
 
     version('master', branch='master')
+    version('0.3.0',      sha256='5ce38eb5b3c2b394bca1316310758f276c893dd3f4c15d7bc14ea05d3110ce58', url='https://github.com/hpc/libcircle/releases/download/v0.3/libcircle-0.3.0.tar.gz')
```
Comment (gonsie) [inline var/spack/repos/builtin/packages/libcircle/package.py]:
```suggestion
    version('0.3.0',      sha256='5ce38eb5b3c2b394bca1316310758f276c893dd3f4c15d7bc14ea05d3110ce58')
```

File: var/spack/repos/builtin/packages/libcircle/package.py [hunk 2]
```diff
@@ -15,6 +15,7 @@ class Libcircle(AutotoolsPackage):
     url      = "https://github.com/hpc/libcircle/releases/download/0.2.1-rc.1/libcircle-0.2.1-rc.1.tar.gz"
```
Comment (gonsie) [inline var/spack/repos/builtin/packages/libcircle/package.py]:
```suggestion
     url      = "https://github.com/hpc/libcircle/releases/download/v0.3/libcircle-0.3.0.tar.gz"
```

Comment (adammoody) [reply var/spack/repos/builtin/packages/libcircle/package.py]: Thanks @gonsie , yes, actually I tried something similar to your suggestion as the first shot.  I only added v0.3 with the version and the sha256 code.  However, then it failed to find the package when downloading.  It worked when I added the url field, as well.  My guess i that between adding the leading "v" character and also using a tag of 0.3 for a package of 0.3.0, the normal spack logic to compute the URL failed.

We could update the package-wide URL to use the new v0.3, but then my worry is that the 0.2.1 version will not download successfully.  My guess is that we'll need to use a url field on something here.  If that's true, I guess we could update the package-wide URL for the v0.3 and also update the 0.2.1 line to specify it's own url if needed.

Comment (adammoody) [reply var/spack/repos/builtin/packages/libcircle/package.py]: We could perhaps also change the github tag to v0.3.0, but I suspect that's only half the problem, with the new leading "v" being the other problem.

**Comment by adammoody** [comment]
To be clear, here is what I tried first
```
    url      = "https://github.com/hpc/libcircle/releases/download/0.2.1-rc.1/libcircle-0.2.1-rc.1.tar.gz"

    version('master', branch='master')
    version('0.3.0',      sha256='5ce38eb5b3c2b394bca1316310758f276c893dd3f4c15d7bc14ea05d3110ce58')
    version('0.2.1-rc.1', sha256='5747f91cf4417023304dcc92fd07e3617ac712ca1eeb698880979bbca3f54865')
```
However, that failed to fetch the v0.3 tarball.  I think spack was computing what it though the URL should be for v0.3 based on the URL format for 0.2.1-rc1.  However, I suspect that broke because of the newly added leading "v" character, and the fact that the middle part is using "v0.3" and not "v0.3.0".

It probably would work up if we change the github tag to be "v0.3" to "0.3.0", or if decide to keep the existing "v0.3" tag, then I think we'll have to set the URL here for either 0.3 or 0.2.1 because of the change.

What do you think?

**Pull request merged**